src/hooks/use-authentication.ts

```diff
@@ -0,0 +1,46 @@
+import { useSession, signIn, signOut } from 'next-auth/react'
+import { SessionData, UserData, ValidAuthProviderId } from 'types/auth'
+
+/**
+ * A minimal wrapper around next-auth/react's `signIn` function. Purpose is to
+ * handle invoking the wrapped function with a default value.
+ */
+const signInWrapper = (
+	provider: ValueOf<typeof ValidAuthProviderId> = ValidAuthProviderId.CloudIdp
```

Comment:
suggestion: We can simplify this type to either specify the enum itself:
```suggestion
	provider: ValidAuthProviderId = ValidAuthProviderId.CloudIdp
```

or a union of all of the enum's values ([ref](https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time)):
```suggestion
	provider: keyof typeof ValidAuthProviderId = ValidAuthProviderId.CloudIdp
```